scylla/src/transport/connection_pool.rs

```diff
@@ -80,6 +90,43 @@ enum PoolConnections {
     },
 }
 
+fn format_connection_vector(conns: &[Arc<Connection>]) -> String {
+    conns
+        .iter()
+        .map(|conn| conn.get_connect_address().to_string())
+        .collect::<Vec<String>>()
+        .join(",")
```

Comment:
Won't the addresses always be either "ip:non-shard-aware-port" or "ip:shard-aware-port"? Maybe we could group by the address (if the IPs are the same we could only print ports) and report how many connections are there to each address?

Reply:
Assuming there are no bugs - yeah, sounds reasonable. But here I'm interested in raw data, in case we keep bogus contents in `connections`, so the less processed it is, the better

Reply:
Ah, now I see that this function prints per-shard connections. Grouping by addresses won't reduce verbosity for Scylla connections unless somebody has more than one connection per shard, which sounds rare. Maybe it will help Cassandra users because they are treated as if there was only one shard, but I won't insist too much on implementing it.